client/rest-high-level/src/main/java/org/elasticsearch/client/security/AuthenticateResponse.java

```diff
@@ -114,24 +124,35 @@ public String getAuthenticationType() {
         return authenticationType;
     }
 
+    public Map<String, Object> getToken() {
```

Comment:
Should this be some new (nested?) class rather than a `Map` ?

I'm of 2 minds
- on one hand, it's hardly ever going to be used, so simple seems better
- but, really it should be a type like (almost) everything else in the HLRC.

Reply:
I just [made the change](https://github.com/ywangd/elasticsearch/commit/791988e02bb005511d36ba2b9a3a8e6723b6c10e) to replace the `Map` with a nested `TokenInfo` class. But now I still think it might be better for it to be just a `Map` because this field is more of the metadata for the authentication type field. Also its own fields may not all exist or make sense for different token types if we decide to expand it to cover other types. For example, `name` does not apply to OAuth Tokens. So it needs to be declared as `StringOrNull`. Overall I think its simpler for it to be just a `Map`.

Reply:
In case it is not clear from the last comment: I am keeping the `token` field as a `Map`. The change is in another branch that I decided to **not** merge.

Reply:
Sorry to drag on this minor issue, but I decided to make one small change: When the response does not have a `token` field, instead of an empty `Map`, the code now represents it just as a `null` value. I think a `null` value is more accurate semantically. Also this would be the case if we replace the `Map` with a named class. So overall I feel `null` is better suited.